Mathlib/Tactic/LiftLets.lean

```diff
@@ -21,6 +21,10 @@ This list is used during the computation to merge let bindings.
 -/
 private partial def Lean.Expr.liftLetsAux (e : Expr) (fvars : Array Expr)
     (f : Array Expr → Expr → MetaM Expr) : MetaM Expr := do
+  if (e.find? Expr.isLet).isNone then
+    return ← f fvars e
+  if ← Meta.isProof e then
+    return ← f fvars e
```

Comment:
Possibly worth a brief comment that looking at proof is pointless as we don't care about the data within them.

Reply:
I was just about to revert this one -- it's probably actually occasionally useful being able to lift lets out of proofs. Though maybe it should be a configuration option?

Reply:
This seems pretty useless to me; any data witness in a proof is irrelevant to unification, and any other attempt to access the data is probably an indication of an API failure. I don't mind making it configurable, but I think the default should be not to.